pkg/operator/awsTagController.go

```diff
@@ -0,0 +1,324 @@
+				return
+			}
+			newInfra, ok := cur.(*configv1.Infrastructure)
+			if !ok || newInfra == nil {
+				return
+			}
+			if oldInfra.Status.PlatformStatus != nil && oldInfra.Status.PlatformStatus.AWS != nil &&
+				newInfra.Status.PlatformStatus != nil && newInfra.Status.PlatformStatus.AWS != nil {
+				if !reflect.DeepEqual(oldInfra.Status.PlatformStatus.AWS.ResourceTags, newInfra.Status.PlatformStatus.AWS.ResourceTags) {
+					c.queue.Add(workQueueKey)
+					return
+				}
+			}
+		},
+	}
+}
+
+// Run is the main method for starting the AWS controller
+func (c *AWSTagController) Run(ctx context.Context) {
+	defer k8sruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	klog.Infof("Starting AWS Controller")
+	if !cache.WaitForCacheSync(ctx.Done(), c.cachesToSync...) {
+		return
+	}
+
+	go wait.Until(c.runWorker, time.Second, ctx.Done())
+
+	klog.Infof("Started AWS Controller")
+	<-ctx.Done()
+	klog.Infof("Shutting down AWS Controller")
+}
+
+func (c *AWSTagController) runWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+// processNextWorkItem is for prcessing the event received
+// which blocks until a new item is received
+func (c *AWSTagController) processNextWorkItem() bool {
+	obj, shutdown := c.queue.Get()
+	if shutdown {
+		return false
+	}
+	defer c.queue.Done(obj)
+
+	klog.V(5).Infof("AWSController: got event from workqueue")
+	if err := c.sync(); err != nil {
+		c.queue.AddRateLimited(workqueueKey)
+		klog.Errorf("AWSController: failed to process event: %s, requeuing", err)
+	} else {
+		c.queue.Forget(obj)
+		klog.V(5).Infof("AWSController: event from workqueue successfully processed")
+	}
+	return true
+}
+
+// sync method is defined for handling the operations required
+// on receiving a informer event.
+// Fetches image registry config data, required for obtaining
+// the S3 bucket configuration and creating a driver out of it
+func (c *AWSTagController) sync() error {
+	return c.syncTags()
+}
+
+// syncTags fetches user tags from Infrastructure resource, which
+// is then compared with the tags configured for the created S3 bucket
+// fetched using the driver object passed and updates if any new tags.
+func (c *AWSTagController) syncTags() error {
+	cr, err := c.imageRegistryConfigClient.Get(
+		context.Background(),
+		defaults.ImageRegistryResourceName,
+		metav1.GetOptions{},
+	)
+	if err != nil {
+		return err
+	}
+
+	// if s3 storage config is missing, must be
+	// non-AWS platform, so not treating it as error
+	if cr.Spec.Storage.S3 == nil {
+		return nil
+	}
+
+	// make a copy to avoid changing the cached data
+	cr = cr.DeepCopy()
+
+	infra, err := c.infraConfigClient.Get(
+		context.Background(),
+		defaults.InfrastructureResourceName,
+		metav1.GetOptions{},
+	)
+	if err != nil {
+		klog.Errorf("failed to fetch Infrastructure resource: %v", err)
+		return err
+	}
+
+	// tags deletion is not supported. Should the user remove it from
+	// PlatformStatus will be looked up for retaining the tag
```

Comment:
I don't really understand the second sentence in this comment:
> Should the user remove it from PlatformStatus will be looked up for retaining the tag

Could you review and perhaps rephrase it?